### ISSUE TITLE
[Bug] AddSignIn leads to exception if Host is not reachable

### ISSUE THREAD

**Issue by ManuelHaas** [AzureAD/microsoft-identity-web#135]
**Which Version of Microsoft Identity Web are you using ?**
Microsoft Identity Web 0.1.1-preview

If the defined host is not reachable while startup AddSignIn leads to an exception and application startup fails. Try the provided Sample Application with disabled network connection.

The "traditional" code `services.AddAuthentication(AzureADDefaults.AuthenticationScheme).AddAzureAD(...)` has no problem with this situation.

The provided sample:
[AzureAd_Test_WebApplication.zip](https://github.com/AzureAD/microsoft-identity-web/files/4550319/AzureAd_Test_WebApplication.zip)

```
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware: Error: An unhandled exception has occurred while executing the request.
System.InvalidOperationException: IDX20803: Unable to obtain configuration from: '[PII is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
 ---> System.IO.IOException: IDX20804: Unable to retrieve document from: '[PII is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
 ---> System.Net.Http.HttpRequestException: Der angegebene Host ist unbekannt.
 ---> System.Net.Sockets.SocketException (11001): Der angegebene Host ist unbekannt.
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.DiagnosticsHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.IdentityModel.Protocols.HttpDocumentRetriever.GetDocumentAsync(String address, CancellationToken cancel)
   --- End of inner exception stack trace ---
   at Microsoft.IdentityModel.Protocols.HttpDocumentRetriever.GetDocumentAsync(String address, CancellationToken cancel)
   at Microsoft.Identity.Web.InstanceDiscovery.IssuerConfigurationRetriever.GetConfigurationAsync(String address, IDocumentRetriever retriever, CancellationToken cancel)
   at Microsoft.IdentityModel.Protocols.ConfigurationManager`1.GetConfigurationAsync(CancellationToken cancel)
   --- End of inner exception stack trace ---
   at Microsoft.IdentityModel.Protocols.ConfigurationManager`1.GetConfigurationAsync(CancellationToken cancel)
   at Microsoft.IdentityModel.Protocols.ConfigurationManager`1.GetConfigurationAsync()
   at Microsoft.Identity.Web.Resource.AadIssuerValidator.GetIssuerValidator(String aadAuthority)
   at Microsoft.Identity.Web.WebAppAuthenticationBuilderExtensions.<>c__DisplayClass1_0.<AddSignIn>b__0(OpenIdConnectOptions options)
   at Microsoft.Extensions.Options.ConfigureNamedOptions`1.Configure(String name, TOptions options)
   at Microsoft.Extensions.Options.OptionsFactory`1.Create(String name)
   at Microsoft.Extensions.Options.OptionsMonitor`1.<>c__DisplayClass11_0.<Get>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Extensions.Options.OptionsCache`1.GetOrAdd(String name, Func`1 createOptions)
   at Microsoft.Extensions.Options.OptionsMonitor`1.Get(String name)
   at Microsoft.AspNetCore.Authentication.AuthenticationHandler`1.InitializeAsync(AuthenticationScheme scheme, HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationHandlerProvider.GetHandlerAsync(HttpContext context, String authenticationScheme)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.MigrationsEndPointMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.DatabaseErrorPageMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**Comment by jennyf19** [comment]
@ManuelHaas Thanks for the detailed info and repro. I tried with a web app sample using Identity web, a `dotnet new --auth SingleOrg` which uses net core auth, and your repro both with `.AddAzureAD(...)` and `.AddSignIn(...)`. I was unable to get any of these to work with a disabled network connection. I got the same error in all the above situations, which i believe is the same one you are getting? 
Could you provide a short clip or screen shots of the experience you get with `.AddAzureAD(...)`? You can email them if you prefer: jeferrie@microsoft.com 
Thank you.
![image](https://user-images.githubusercontent.com/19942418/85097209-a987e180-b1ab-11ea-9109-61176e09509e.png)

**Comment by jennyf19** [comment]
@ManuelHaas Thanks for the detailed info and repro. I tried with a web app sample using Identity web, a dotnet new --auth SingleOrg which uses net core auth, and your repro both with .AddAzureAD(...) and .AddSignIn(...). I was unable to get any of these to work with a disabled network connection. I got the same error in all the above situations, which i believe is the same one you are getting?
![image](https://user-images.githubusercontent.com/19942418/85161377-834a5c00-b214-11ea-97d6-cc7ba953cbe5.png)

so in looking at this more, it appears that the issue is actually related to the other [issue](https://github.com/AzureAD/microsoft-identity-web/issues/133) you opened and is a scenario we do not yet support. I would recommend closing as a duplicate. your thoughts?
